src/Microsoft.OData.Client/BulkUpdateSaveResult.cs

```diff
@@ -99,11 +101,70 @@ protected override Stream ResponseStream
             get { return this.responseStream; }
         }
 
+        /// <summary>
+        /// Asynchronous bulk update request.
+        /// </summary>
+        /// <typeparam name="T">The type of the top-level objects to be deep-updated.</typeparam>
+        /// <param name="objects">The top-level objects of the type to be deep updated.</param>
+        internal void BeginBulkUpdateRequest<T>(params T[] objects)
+        {
+            PerRequest pereq = null;
+
+            if (objects == null || objects.Length == 0)

```

Comment:
why can't check the input at the beginning?

Reply:
We do check at the beginning. This is an internal method though and can be called from other areas of the project.